app/src/main/java/com/example/taskyy/data/repositories/AuthRepositoryImpl.kt

```diff
@@ -53,32 +56,48 @@ class AuthRepositoryImpl @Inject constructor(
         return hasLowerCase && hasUpperCase && hasDigit && hasValidLength
     }
 
-    override suspend fun login(login: Login): com.example.taskyy.domain.error.Result<User, DataError.Network> {
+    override suspend fun login(login: Login): Result<User, DataError.Network> {
         return try {
             val loginUser = retrofit.loginUser(login)
             val user = loginUser.toUser()
             val userEntity = loginUser.toUserEntity(login.email, loginUser.userId)
             userDao.insertUser(userEntity)
             addTokenAndIdToDatabase(loginUser.token, loginUser.userId, login.email)
-            com.example.taskyy.domain.error.Result.Success(user)
+            Result.Success(user)
         } catch (e: HttpException) {
             when (e.code()) {
-                408 -> com.example.taskyy.domain.error.Result.Error(DataError.Network.REQUEST_TIMEOUT)
-                409 -> com.example.taskyy.domain.error.Result.Error(DataError.Network.INCORRECT_PASSWORD_OR_EMAIL)
-                429 -> com.example.taskyy.domain.error.Result.Error(DataError.Network.TOO_MANY_REQUESTS)
-                413 -> com.example.taskyy.domain.error.Result.Error(DataError.Network.PAYLOAD_TOO_LARGE)
-                500 -> com.example.taskyy.domain.error.Result.Error(DataError.Network.SERVER_ERROR)
-                400 -> com.example.taskyy.domain.error.Result.Error(DataError.Network.SERIALIZATION)
-                else -> com.example.taskyy.domain.error.Result.Error(DataError.Network.UNKNOWN)
+                408 -> Result.Error(DataError.Network.REQUEST_TIMEOUT)
+                409 -> Result.Error(DataError.Network.INCORRECT_PASSWORD_OR_EMAIL)
+                429 -> Result.Error(DataError.Network.TOO_MANY_REQUESTS)
+                413 -> Result.Error(DataError.Network.PAYLOAD_TOO_LARGE)
+                500 -> Result.Error(DataError.Network.SERVER_ERROR)
+                400 -> Result.Error(DataError.Network.SERIALIZATION)
+                else -> Result.Error(DataError.Network.UNKNOWN)
             }
         } catch (e: IOException) {
             when (e.message) {
-                "UnknownHostException" -> com.example.taskyy.domain.error.Result.Error(DataError.Network.UNKNOWN_HOST_EXCEPTION)
-                else -> com.example.taskyy.domain.error.Result.Error(DataError.Network.UNKNOWN)
+                "UnknownHostException" -> Result.Error(DataError.Network.UNKNOWN_HOST_EXCEPTION)
+                else -> Result.Error(DataError.Network.UNKNOWN)
             }
         }
     }
 
+    override suspend fun validateToken(): Boolean {
+        val call = retrofit.checkTokenIsValid()
+        var returnValue = false
+        call.enqueue(object : Callback<Void> {
```

Comment:
If you're using suspend functions, you don't need enqueue. You can make your Retrofit call return the response directly, without wrapping it into a `Call` object.